Wit/WITRecorder.m

```diff
@@ -46,28 +47,45 @@ static void audioQueueInputCallback(void* data,
     int err;
     
     if (WIT_DEBUG) {
-        debug(@"Audio chunk %u/%u", (unsigned int)size, (unsigned int)buffer->mAudioDataBytesCapacity);
+        debug(@"Audio chunk %u/%u, numofpacdescriptions %d", (unsigned int)size, (unsigned int)buffer->mAudioDataBytesCapacity, numberPacketDescriptions);
     }
-
+    
     if (size > 0) {
         NSData* audio = [NSData dataWithBytes:bytes length:size];
         @autoreleasepool {
             WITRecorder* recorder = (__bridge WITRecorder*)data;
-            [recorder.delegate recorderGotChunk:audio];
-            if (recorder.vad != nil) {
+            if (recorder.delegate && recorder.isRecording == YES) {
+                [recorder.delegate recorderGotChunk:audio];
+            }
+            
+            if (recorder.vad != nil && recorder.isRecording == YES) {
                 [recorder.vad gotAudioSamples:audio];
             }
+            if (recorder.isRecording == YES) {
+                err = AudioQueueEnqueueBuffer(q, buffer, 0, NULL);
+                if (err) {
+                    debug(@"Error when enqueuing buffer from callback: %d", err);
+                }
+            }
```

Comment:
This could be summarized as:

```
if (recorder.isRecording) {
  [recorder.delegate recorderGotChunk:audio];
  [recorder.vad gotAudioSamples:audio];

  err = AudioQueueEnqueueBuffer(q, buffer, 0, NULL);
  if (err) {
    debug(@"Error when enqueuing buffer from callback: %d", err);
  }
}